packages/databyss-editor/lib/markup.js

```diff
@@ -25,7 +24,12 @@ export const applyRange = (editor, range) => {
   // and correct the offset with an additional move, below.
   Transforms.move(editor, { distance: 1, edge: 'anchor', reverse: true })
 
-  toggleMark(editor, range.mark)
+  // apply marks array
+  if (range.marks) {
```

Comment:
I think we should keep the `range` element to a single mark, because that allows metadata to be added specific to that mark, and keeps our data structures from getting too deeply nested.